tpot/tpot.py

```diff
@@ -1135,6 +1141,210 @@ def _rbf(self, input_df, gamma, n_components):
+        training_features = input_df.loc[input_df['group'] == 'training'].drop(self.non_feature_columns, axis=1)
+
+        if len(training_features.columns.values) == 0:
+            return input_df.copy()
+
+        if n_clusters < 1:
+            n_clusters = 1
+
+        affinity_types = ['euclidean', 'l1', 'l2', 'manhattan', 'cosine', 'precomputed']
+        linkage_types = ['ward', 'complete', 'average']
+
+        linkage_name = linkage_types[linkage % len(linkage_types)]
+
+        if linkage_name == 'ward':
+            affinity_name = 'euclidean'
+        else:
+            affinity_name = affinity_types[affinity % len(affinity_types)]
+
+        fa = FeatureAgglomeration(n_clusters=n_clusters, affinity=affinity_name, linkage=linkage_name)
+        fa.fit(training_features.values.astype(np.float64))
+
+        clustered_features = fa.transform(input_df.drop(self.non_feature_columns, axis=1).values.astype(np.float64))
+
+        modified_df = pd.DataFrame(data=clustered_features)
+        modified_df['class'] = input_df['class'].values
+        modified_df['group'] = input_df['group'].values
+        modified_df['guess'] = input_df['guess'].values
+
+        new_col_names = {}
+        for column in modified_df.columns.values:
+            if type(column) != str:
+                new_col_names[column] = str(column).zfill(10)
+        modified_df.rename(columns=new_col_names, inplace=True)
+
+        return modified_df.copy()
+
+    def _nystroem(self, input_df, kernel, gamma, n_components):
+        """
+        Uses scikit-learn's Nystroem to approximate a kernel map using a subset
+        of the training data. Constructs an approximate feature map for an
+        arbitrary kernel using a subset of the data as basis.
+
+        Parameters
+        ----------
+        input_df: pandas.DataFrame {n_samples, n_features+['class', 'group', 'guess']}
+            Input DataFrame to scale
+        kernel: int
+            Kernel type is selected from scikit-learn's provided types:
+                'sigmoid', 'polynomial', 'additive_chi2', 'poly', 'laplacian', 'cosine', 'linear', 'rbf', 'chi2'
+
+            Input integer is used to select one of the above strings.
+        gamma: float
+            Gamma parameter for the kernels.
+        n_components: int
+            The number of components to keep
+
+        Returns
+        -------
+        modified_df: pandas.DataFrame {n_samples, n_components + ['guess', 'group', 'class']}
+            Returns a DataFrame containing the transformed features
+
+        """
+        training_features = input_df.loc[input_df['group'] == 'training'].drop(self.non_feature_columns, axis=1)
+
+        if len(training_features.columns.values) == 0:
+            return input_df.copy()
+
+        if n_components < 1:
+            n_components = 1
+        else:
+            n_components = min(n_components, len(training_features.columns.values))
+
+        kernel_types = list(PAIRWISE_KERNEL_FUNCTIONS.keys())
```

Comment:
I think we should hard-code this list as we do in `_feat_agg()`. I'm concerned that the order may change in future versions of sklearn, and that could mess up our encoding here. Instead, let's take the current values/ordering from `list(PAIRWISE_KERNEL_FUNCTIONS.keys())` and hard-code it.